ex2_system/source/system_state/state_task.c

```diff
@@ -173,79 +181,33 @@ SYS_returnState power_switch_uhf(bool *uhf_status) {
               1);  // Even if it was not set to zero in this function.
     uint8_t is_low_pwr;
     status = HAL_UHF_getLowPwr(&is_low_pwr);
-    if (uhf_status == 0) {
+    if (uhf_status == false) {
       if (is_low_pwr) {
         // send any command to wake it up, e.g.,
         uint8_t scw[12];
-        status = HAL_UHF_getSCW(&scw);
+        status = HAL_UHF_getSCW(scw);
         // Here write any code based on the read scw!
       }
     }
-    if (status != 0) {
-      printf("Switching UHF TRX's power failed.");
+    if (status < 0) {
+      ex2_log("Switching UHF TRX's power failed.");
       // An error occurred. Call this function again?
+      return;
     }
   }
-  return status;
-}
-
-/*
- * Independent functions are considered for easier modification.
- * e.g., We might need to initiate a procedure for a subsystem before
- * disconnecting power.
- */
```

Comment:
The idea of having different functions was to later add more stuff for specific subsystems prior to turning them off.

Reply:
Yeah I recall that. What would you think about moving these into the subsystem HAL folders? These functions don't need to be here I don't think since we're passing the target state so we can do any error reporting in the HALs

Reply:
Sounds good to me. Could you create a Github issue for that once you merge this pr?

Reply:
Yeah, will do!